File: src/api/routes/deployment.py
"""Deployment configuration API routes (KMFLOW-7).

Exposes deployment capabilities and health information so the frontend
can adapt its UI based on what features are available (e.g. hide copilot
when no LLM is configured, show data residency warnings).
"""

from __future__ import annotations

from typing import Any

from fastapi import APIRouter, Depends
from pydantic import BaseModel

from src.api.middleware.data_residency import get_deployment_capabilities
from src.core.auth import get_current_user
from src.core.models import User

router = APIRouter(prefix="/api/v1/deployment", tags=["deployment"])


class DeploymentCapabilitiesResponse(BaseModel):
    """Response schema for deployment capabilities."""

    llm_enabled: bool = False
    neo4j_enabled: bool = False
    minio_enabled: bool = False
    data_residency: str = "none"
    environment: str = "unknown"


@router.get("/capabilities", response_model=DeploymentCapabilitiesResponse)
async def capabilities(
    current_user: User = Depends(get_current_user),
) -> dict[str, Any]:
    """Return deployment capabilities.

    Used by the frontend to determine which features are available
    in this deployment (cloud vs. on-prem vs. air-gapped).
    """
    return get_deployment_capabilities()

File: frontend/src/lib/api/deployment.ts
/**
 * Deployment capabilities API client (KMFLOW-7).
 *
 * Fetches deployment configuration to adapt the UI based on
 * which features are available (cloud vs on-prem vs air-gapped).
 */

import { apiGet } from "./client";

export interface DeploymentCapabilities {
  llm_available: boolean;
  llm_provider: string;
  llm_is_local: boolean;
  embeddings_local: boolean;
  data_residency_default: string;
  copilot_enabled: boolean;
  scenario_suggestions_enabled: boolean;
  gap_rationale_enabled: boolean;
}

/**
 * Fetch deployment capabilities from the API.
 * Used to conditionally render LLM-dependent features.
 */
export async function getDeploymentCapabilities(): Promise<DeploymentCapabilities> {
  return apiGet<DeploymentCapabilities>("/api/v1/deployment/capabilities");
}
